react/features/video-layout/subscriber.js

```diff
@@ -111,7 +115,13 @@ function _updateAutoPinnedParticipant({ dispatch, getState }) {
 
     if (latestScreenshareParticipantId) {
         dispatch(pinParticipant(latestScreenshareParticipantId));
-    } else if (getPinnedParticipant(state['features/base/participants'])) {
+
+        // If we are in audio-only mode, set last N to 1.
+        appState === 'active' && audioOnly && dispatch(setLastN(1));
```

Comment:
Won't this leave lastN to 1 even if a non-screenshare participant is pinned?

Reply:
Yes, but note that 2 lines above we pin the screen-sharing participant. The plan is to also "lock" the pinning capability...